### PR TITLE
Update (redis) leaderboard cache when modifying restriction status of a player

### LINKED ISSUE #172: After restriction player is still in redis
There's not too much to explain. After player restriction player is still in redis.

### PR DESCRIPTION
**Pull request by alowave223**
fixes #172

### CODE REVIEW

File: app/objects/player.py [hunk 1]
```diff
@@ -522,6 +532,20 @@ async def unrestrict(self, admin: "Player", reason: str) -> None:
             {"from": admin.id, "to": self.id, "action": "unrestrict", "msg": reason},
         )
 
+        if not self.stats:
+            async with app.state.services.database.connection() as db_conn:
+                await self.stats_from_sql_full(db_conn)
```
Comment (cmyui) [inline app/objects/player.py:537]: is this case possible?

Comment (alowave223) [reply app/objects/player.py:537]: if player not online

File: app/objects/player.py [hunk 2]
```diff
@@ -522,6 +532,20 @@ async def unrestrict(self, admin: "Player", reason: str) -> None:
...
+
+        for idx, mode_stat in enumerate(self.stats.values()):
```
Comment (cmyui) [inline app/objects/player.py:539]: stats keys are modes i believe, could we do `for mode, stats in self.stats.items():`? (don't have time to check atm, at work)

**Comment by alowave223** [comment]
Should be fine now, i guess.

**Comment by cmyui** [comment]
is the code tested & working? if you're not able to test, i can after work

File: app/objects/player.py [hunk 3]
```diff
@@ -495,6 +495,16 @@ async def restrict(self, admin: "Player", reason: str) -> None:
             {"from": admin.id, "to": self.id, "action": "restrict", "msg": reason},
         )
 
+        for mode in range(8):
```
[the remark this answers is not in the document]
Comment (cmyui) [reply app/objects/player.py]: lol i should really make a tuple of these somewhere so it can be re-used

**Comment by alowave223** [comment]
> is the code tested & working? if you're not able to test, i can after work

my gulag instance won't startup because of #174

**Comment by cmyui** [comment]
> > is the code tested & working? if you're not able to test, i can after work
> 
> my gulag instance won't startup because of #174

try `git rm oppai-ng`

**Comment by cmyui** [comment]
tested & gucci

**Pull request merged**